apollo-gradle-plugin-incubating/src/main/kotlin/com/apollographql/apollo/gradle/api/Service.kt

```diff
@@ -1,60 +1,23 @@
 package com.apollographql.apollo.gradle.api
 
-import com.apollographql.apollo.gradle.api.Introspection
+import com.apollographql.apollo.gradle.internal.DefaultIntrospection
 import groovy.lang.Closure
 import org.gradle.api.Action
+import org.gradle.api.provider.ListProperty
+import org.gradle.api.provider.Property
 
-class Service(val name: String): CompilerParams by DefaultCompilerParams() {
-  /**
-   * Place where the schema.json file is.
-   * This path is relative to the current project directory.
-   * schemaFilePath can point outside of src/{foo}/graphql but in that case, you'll want
-   * to define rootPackageName else fragments/types will be stored at the root of the namespace.
-   * By default, the plugin will look for a schema.json file in src/{foo}/graphql
-   */
-  var schemaFilePath: String? = null
+interface Service: CompilerParams {
+  fun introspection(configure: Action<in Introspection>)
 
-  /**
-   * Place where the graphql files are searched.
-   * This path is relative to the current sourceSet (e.g src/{foo}/graphql/{sourceFolderPath})
-   * By default, this is the directory where the schema.json is stored or "." if the schema is outside.
-   * You need to define this if you have two or more services whose schema is stored outside src/{foo}/graphql.
-   * If not, you can certainly omit it.
-   */
-  var sourceFolderPath: String? = null
+  val schemaPath: Property<String>
```

Comment:
Similar to my comment below, this could be better to be `RegularFileProperty`. I found myself setting this as file all the time.